### ISSUE TITLE
completion execute command.

### ISSUE THREAD

**Issue by Carreau** [xonsh/xonsh#1694]
A (limited) case of completion do execute commands. 

For example have `mvim` as a script to send the current file to the mac vim server.

```
$ which mvim
/Users/bussonniermatthias/bin/mvim
```
```
 $ mvim .tra<tab>
```

Trigger macvim to open the (non existing) file `.tra mvim`... 

Bisecting.

**Comment by Carreau** [comment]
That seem to be because of 188fee34dd13dc44f4122c5067108d208b8c6b42 part of https://github.com/xonsh/xonsh/pull/1602 by @blahgeek I don't see any obvious reason why it would do that...

**Comment by Carreau** [comment]
Here is the BASH_COMPLETION_SCRIPT filled in with variables:

```
source "/usr/local/etc/bash_completion"
if (complete -p "mvim" 2> /dev/null || echo _minimal) | grep --quiet -e "_minimal"
then
    declare -f _completion_loader > /dev/null && _completion_loader "mvim"
fi
_func=$(complete -p mvim | grep -o -e '-F \w\+' | cut -d ' ' -f 2) 
COMP_WORDS=(mvim .tra)
COMP_LINE='mvim .tra'
COMP_POINT=${#COMP_LINE}
COMP_COUNT=10
COMP_CWORD=1
$_func mvim .tra mvim
for ((i=0;i<${#COMPREPLY[*]};i++)) do echo ${COMPREPLY[i]}; done
```

This `complete -p mvim ` complains: `bash: complete: mvim: no completion specification` and then `$_func mvim .tra mvim` triggers vim to open.

**Comment by Carreau** [comment]
In master it just seem like `"$_complete_stmt"` is the empty string.

**Comment by blahgeek** [comment]
Thanks for finding this! I would take a look at it later today. Sorry for the bug

**Comment by blahgeek** [comment]
It seems that it only happens with bash-completion version 1.x. For the moment you could try installing bash-completion 2.x (if you are using homebrew on macOS, you could run `brew unlink bash-completion && brew install bash-completion2`). Please let me know if it works. Thanks!

**Comment by xsteadfastx** [comment]
i run into the same problem right now.  problem is: i dont have a bash-completion 2 package for ubuntu :)

**Comment by blahgeek** [comment]
hi @xsteadfastx @Carreau sorry for the bug, I created a PR #1701 , it would be great if you can test it and give me some feedbacks. Thanks!

**Comment by xsteadfastx** [comment]
@blahgeek looks fine to me. i just did a quick check.

**Comment by gforsyth** [comment]
Resolved in #1701.  Please reopen if the issue persists!
